sdk/core/azure-core/azure/core/pipeline_client.py

```diff
@@ -118,24 +118,35 @@ def _request(self, method, url, params, headers, content, form_content, stream_c
 
         return request
 
-    def close(self):
-        self.__exit__()
+    def _format_url_section(self, template, **kwargs):
+        while True:
+            try:
+                return template.format(**kwargs)
+            except KeyError as key:
+                components = template.split("/")
+                new_components = [c for c in components if "{{{}}}".format(key.args[0]) not in c]
+                template = "/".join(new_components)
```

Comment:
components only ever shrinks from a fixed initial size, correct? It's probably a past life in more embedded like systems but I would suggest trying to recast this as a finite loop if at all possible